src/card_operations/check_pairing.c

```diff
@@ -0,0 +1,147 @@
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "check_pairing.h"
+
+#include "card_internal.h"
+#include "ui_delay.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * TODO: add description when this api gets concrete definition
+ *
+ * @param connection_data
+ */
+static void init_tap_card_data(NFC_connection_data *connection_data);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+static void init_tap_card_data(NFC_connection_data *connection_data) {
+  memset(connection_data, 0, sizeof(NFC_connection_data));
+  connection_data->retries = 5;
+  connection_data->desktop_control = false;
+  connection_data->acceptable_cards = ACCEPTABLE_CARDS_ALL;
+  memset(connection_data->family_id, 0xff, sizeof(connection_data->family_id));
+  memset(connection_data->card_key_id, 0, sizeof(connection_data->card_key_id));
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+card_error_type_e card_check_pairing(check_pairing_result_t *result) {
+  LOG_SWV("%s (%d): %p\n", __func__, __LINE__, result);
+  card_error_type_e status = CARD_OPERATION_DEFAULT_INVALID;
+  if (NULL == result) {
+    return status;
+  }
+
+  memset(result, 0, sizeof(check_pairing_result_t));
+  result->card_number = 0;
+  card_operation_data_t operation_data = {tap_card_data, NULL, 0};
+  init_tap_card_data(&operation_data.nfc_data);
```

Comment:
Here we are first initializing with tap_card_data then resetting it. Is tap_card_data left by mistake? As the nfc_data member is not a pointer

Reply:
removed